subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/internal/ValidateAction.java

```diff
@@ -100,24 +103,50 @@ public void visitFile(FileVisitDetails fileDetails) {
 
     private static void collectValidationProblems(Class<?> topLevelBean, Map<String, Boolean> problems, boolean enableStricterValidation) {
         boolean cacheable;
-        boolean mapErrorsToWarnings;
+        boolean canBeMadeCacheable;
+        boolean treatWarningsAsErrors;
         DocumentationRegistry documentationRegistry = new DocumentationRegistry();
         if (Task.class.isAssignableFrom(topLevelBean)) {
+            canBeMadeCacheable = true;
             cacheable = enableStricterValidation || topLevelBean.isAnnotationPresent(CacheableTask.class);
```

Comment:
Why do we treat tasks as cacheable if strict validation is enabled I wonder? Should `cacheable` be called something else maybe?

Reply:
I renamed `cacheable` to `reportCacheabilityProblems`.